pkg/network/amqp.go

```diff
@@ -71,6 +72,77 @@ func (a *Amqp) Start(started chan bool) {
 	started <- true
 }
 
+// DeclareQueue declare a queue in amqp handler
+func (a *Amqp) DeclareQueue(queueName, exchangeName string) error {
+	err := a.channel.ExchangeDeclare(
+		exchangeName,
+		amqp.ExchangeTopic, // type
+		true,               // durable
+		false,              // delete when complete
+		false,              // internal
+		false,              // noWait
+		nil,                // arguments
+	)
+	if err != nil {
+		a.logger.Error(err)
+		return err
+	}
+
+	queue, err := a.channel.QueueDeclare(
+		queueName,
+		true,  // durable
+		false, // delete when unused
+		false, // exclusive
+		false, // noWait
+		nil,   // arguments
+	)
+	if err != nil {
+		a.logger.Error(err)
+		return err
+	}
+
+	a.queue = &queue
+	return nil
+}
+
+// OnMessage receive messages an put them on channel
+func (a *Amqp) OnMessage(msgChan chan InMsg, queueName, exchange, key string) error {
```

Comment:
Keep the parameter names pattern with DeclareQueue func.

Reply:
@vitbaq up